### PR TITLE
Add ReynoldsNumberFunctorAux

### LINKED ISSUE #20359: Create ReynoldsNumber aux kernel for NS module
## Reason
Reynolds number is a defining quantity of a flow. It's important to know for laminar/turbulence modeling purposes and for stabilization purposes.

## Design
Gonna be based on functors as they're straightforward to use with both FE and FV

## Impact
Users will be able to visualize elemental Reynolds numbers and make decisions about stabilization/turbulence

### LINKED ISSUE #20360: create functor-based scalar magnitude from functor vector components material
## Reason
Lots of use cases. My immediate need is that I need speed for computing the Reynolds number (see #20359 ) and I usually have velocity component variables from finite element or finite volume navier-stokes simulations.

## Design
Retrieve components as functors and compute Euclidean norm

## Impact
Easy computation of vector magnitudes from component quantities

### PR DESCRIPTION
**Pull request by lindsayad**
and `VectorMagnitudeFunctorMaterial`

Closes #20359 and #20360

### CODE REVIEW

File: framework/include/materials/VectorMagnitudeFunctorMaterial.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "FunctorMaterial.h"
+
+/**
+ * This class takes up to three functors corresponding to vector "components and computes the
```
Comment (GiudGiud) [inline framework/include/materials/VectorMagnitudeFunctorMaterial.h:15]:
```suggestion
 * This class takes up to three functors corresponding to vector components and computes the
```

File: modules/navier_stokes/doc/content/source/auxkernels/ReynoldsNumberFunctorAux.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+The `ReynoldsNumberFunctorAux` object computes the element/cell Reynolds number
+given by
+
+\begin{equation}
+Re = \frac{\rho h u}{\mu}
+\end{equation}
+
+where $u$ is the norm of the velocity, e.g. the speed, $h$ the largest dimension
+of the element (computed using `_current_elem->hmax()`), $\rho$ is the density,
```
Comment (GiudGiud) [inline modules/navier_stokes/doc/content/source/auxkernels/ReynoldsNumberFunctorAux.md:15]: technically should probably be h in the direction of velocity
and wouldnt hmin be more limiting?

File: modules/navier_stokes/include/auxkernels/ReynoldsNumberFunctorAux.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AuxKernel.h"
+
+// Forward Declarations
```
Comment (GiudGiud) [inline modules/navier_stokes/include/auxkernels/ReynoldsNumberFunctorAux.h:14]:
```suggestion
```

File: modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  []
+[]
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+  []
+[]
+
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i:158]:
```suggestion
```

File: modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+[Executioner]
+  type = Steady
+  solve_type = 'NEWTON'
+  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_pc_type -sub_pc_factor_shift_type'
+  petsc_options_value = 'asm      100                lu           NONZERO'
+  nl_rel_tol = 1e-12
+  resid_and_jacobian_together = true
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i:165]: is this merged? I did not pay attention :(

File: modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i (hunk 4 above)
Comment (GiudGiud) [reply modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i:158]: isnt that the default for newton with AD?

**Review by GiudGiud — COMMENTED** [review]
want to throw the fix for FunctorFluidProps on Re in here too?

File: modules/navier_stokes/doc/content/source/auxkernels/ReynoldsNumberFunctorAux.md (hunk 2 above)
Comment (lindsayad) [reply modules/navier_stokes/doc/content/source/auxkernels/ReynoldsNumberFunctorAux.md:15]: L is in the numerator so if you're worried about possibly needing turbulence treatment or stabilization then the paranoid estimate is with the largest cell/element L

File: modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i (hunk 5 above)
Comment (lindsayad) [reply modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i:165]: lol I don't know how that snuck in there

Comment (lindsayad) [reply modules/navier_stokes/test/tests/auxkernels/reynolds-number-functor-aux/fv.i:165]:
```suggestion
```

**Comment by lindsayad** [comment]
I really don't understand how those unrelated tests are suddenly failing

File: modules/navier_stokes/doc/content/source/auxkernels/ReynoldsNumberFunctorAux.md (hunk 2 above)
Comment (GiudGiud) [reply modules/navier_stokes/doc/content/source/auxkernels/ReynoldsNumberFunctorAux.md:15]: oh yes my bad

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**